[Raviraj Talgeri] iP

starbucksventi opened the pull request:
# Tesla

>    “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Tesla frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~FAST~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2425s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [x]     Managing tasks
- [x]     Managing deadlines
- [ ]     Reminders (coming soon) 
#
If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        new Duke(FILE_NAME).run();
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

starbucksventi committed:
Add greeting

starbucksventi committed:
Add support for echo

starbucksventi committed:
Add support for list

starbucksventi committed:
Implement mark and unmark tasks

starbucksventi committed:
Implement Todo, Deadline and Event classes

starbucksventi committed:
Implement error handling for incorrect inputs

starbucksventi committed:
Implement delete for tasks

starbucksventi committed:
Add Automated Text UI Testing

starbucksventi committed:
Refactor code, add refined error handling and refine output aesthetics

starbucksventi committed:
Add memory storage capability and perform bug fixes

starbucksventi committed:
Merge branch 'branch-Level-7'

starbucksventi committed:
Add proper date handling for deadline command

starbucksventi committed:
Merge branch 'branch-Level-8'

starbucksventi committed:
Refactor code to implement the objects Ui, Parser, Storage, TaskList

starbucksventi committed:
Merge branch 'branch-AMoreOOP'

starbucksventi committed:
Organize project into a package

starbucksventi committed:
Merge branch 'branch-A-Packages'

starbucksventi committed:
Merge branch 'add-gradle-support'

starbucksventi committed:
Add JUnit tests to test the behaviour of the code

starbucksventi committed:
Merge branch 'branch-A-JUnit'

starbucksventi committed:
Add Jar file creation capability

starbucksventi committed:
Merge branch 'branch-A-Jar'

starbucksventi committed:
Add JavaDoc comments to the code

starbucksventi committed:
Merge branch 'branch-A-Javadoc'

starbucksventi committed:
Edit code to comply with coding standard

starbucksventi committed:
Merge branch 'branch-A-CodingStandard'

starbucksventi committed:
Implement way to find a task by search for a keyword

starbucksventi committed:
Merge branch 'branch-Level-9'

AppleJem commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,30 @@
+package duke;
+
+public class Event extends Task{
```
Would it make sense to insert a docstring here to describe what the Event class does?

AppleJem commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,30 @@
+
+    public Event(String name, String startDate, String endDate) {
+        super(name);
+        setStartDate(startDate);
```
Would it be more readable to set the private fields directly here instead?

AppleJem commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,84 @@
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Arrays;
+
```
Docstring to describe class and methods might make it clearer for future development?

AppleJem commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,84 @@
+        if (inputString.split(" ", 2).length == 0) {
+            throw new EmptyCommandException();
+        }
+        boolean flag1 = !(Arrays.asList("list", "mark", "unmark", "todo", "event", "deadline", "delete")
```
I found it a little hard to read the code here, perhaps insert some comments to explain what's going on?

AppleJem commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,84 @@
+            int idx = Integer.parseInt(remainingInput) - 1;
+            Task taskToBeDeleted = tasks.get(idx);
+            tasks.delete(idx);
+            ui.taskAddOrDeleteDisplay(taskToBeDeleted, "delet", tasks);
```
There seems to be a typo here for "delet"

AppleJem commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,55 @@
+            String[] items = string.split(" , ");
+            String task = items[0];
+            boolean flag = Objects.equals(items[1], "1");
+            if (Objects.equals(task, "T")) {
```
I was thinking it might be a bit more readable to use the `string.equals()` method to compare strings.

AppleJem reviewed (commented):
Maybe can add more comments and docs to explain code for future maintainability!

starbucksventi committed:
Add checkstyle, JavaDoc comments and refactor code

starbucksventi committed:
Merge branch 'branch-A-Checkstyle'

starbucksventi committed:
Implement GUI, add more JavaDoc comments and refactor code

starbucksventi committed:
Merge branch 'branch-Level-10'

starbucksventi committed:
Add varargs capability to matching word search function

starbucksventi committed:
Merge branch 'branch-A-Varargs'

starbucksventi committed:
Implement assertions, improve JUnit testing and add JavaDoc comments

There are several portions where including assertions are included to stop the program should bugs be inadvertently be present.

More JUnit tests for previously untested portions of the code allow bugs to be caught during compilation. Re-delegating some tests to a different unit test class helps to make the testing more intuitive.

Let us add JavaDoc comments to public methods that currently do not have any to ensure the program code abides by Java coding standards.

starbucksventi committed:
Refactor code for readability, add JavaDoc comments and fix checkstyle

Refactoring code, which include extracting methods from larger methods to conform to Single Level Abstraction Principle allows for better readability. Assigning error messages directly to the exception, instead of typing them inividually in the catch block, allows us to standardise the message and set them from a single location, i.e., the specified exception's class.

Let us continue to add JavaDoc comments to the public methods to conform to Java's coding standard.

Import statements also need to match checkstyle as there is a specific order for them to follow.

starbucksventi committed:
Use Java streams to handle matching strings and refactor code

Java streams allow us to work off the pros of functional programming,
namely readability in our context, within a primarily OOP-based code.

Let us use more switch statements in place of long if-else conditional
 statements as this also improves readability.

starbucksventi committed:
Merge pull request #1 from starbucksventi/branch-A-Assertions

Implement assertions, improve JUnit testing and add JavaDoc comments

starbucksventi committed:
Merge branch 'master' of https://github.com/starbucksventi/ip

starbucksventi committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/duke/Duke.java
#	src/main/java/duke/InvalidCommandException.java
#	src/main/java/duke/Parser.java
#	src/main/java/duke/TaskListOutOfBoundsException.java
#	src/main/java/duke/Ui.java
#	src/test/java/duke/ParserTest.java

starbucksventi committed:
Deconflict versions and merge branch 'master' into branch-A-streams

# Conflicts:
#	src/main/java/duke/Parser.java
#	src/test/java/duke/ParserTest.java

starbucksventi committed:
Merge pull request #2 from starbucksventi/branch-A-CodeQuality

Refactor code for readability, add JavaDoc comments and fix checkstyle

starbucksventi committed:
Deconflict and merge branch 'master' into branch-A-streams

# Conflicts:
#	src/main/java/duke/Parser.java

starbucksventi committed:
Merge pull request #3 from starbucksventi/branch-A-streams

Use Java streams to handle matching strings and refactor code

starbucksventi committed:
Add snooze functionality and perform bug fixes

Snooze functionality is added to make the chatbot more versatile and
thus cater to more use cases.

Let us continue to find bug fixes and correct them through repeated
testing of the code for different user inputs as is done in this
commit.

starbucksventi committed:
Merge branch 'branch-BCD-extension'

starbucksventi committed:
Add enhanced GUI features

starbucksventi committed:
Add changes to GUI

starbucksventi committed:
Merge branch 'branch-A-BetterGui'

starbucksventi committed:
Add better personality to the bot

starbucksventi committed:
Merge branch 'branch-A-Personality'

starbucksventi committed:
Add more comprehensive error handling

starbucksventi committed:
Merge branch 'branch-A-MoreTesting'

starbucksventi committed:
Fix JavaDoc comments and delete dead code

starbucksventi committed:
Fix ASCII art in the opening GUI message to the user and refactor code

starbucksventi committed:
Enhance code quality and refactor code using AI tools

starbucksventi committed:
Merge branch 'branch-A-AiAssisted'

starbucksventi committed:
Fix bugs, add title bar to GUI and cite AI usage

starbucksventi committed:
Add User Guide for the product

starbucksventi committed:
Merge branch 'branch-A-UserGuide'

starbucksventi committed:
Enhance code quality to match coding conventions

starbucksventi committed:
Fix grammatical issues in the README

starbucksventi committed:
Add more error handling to improve program usability

starbucksventi committed:
Fix bugs in file storage logic

starbucksventi committed:
Merge branch 'branch-A-MoreErrorHandling'

starbucksventi committed:
Update the README with improved instructions for different OS users

starbucksventi committed:
Add documentation for closing the program

starbucksventi committed:
Fix bugs and add more JavaDoc comments

starbucksventi committed:
Update the README to include the jar file of the product
